src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -27,16 +28,23 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
   options.reconnectionHandler = options.reconnectionHandler || window['Blazor'].defaultReconnectionHandler;
   logger.log(LogLevel.Information, 'Starting up blazor server-side application.');
 
-  const initialConnection = await initializeConnection(options, logger);
-  const circuit = await startCircuit(initialConnection);
+  const components = discoverComponents(document);
+  const circuit = new CircuitDescriptor(components);
+
+  const initialConnection = await initializeConnection(options, logger, circuit);
+  const circuitStarted = await circuit.startCircuit(initialConnection);
+  if (!circuitStarted) {
+    logger.log(LogLevel.Information, 'Failed to start the circuit.');
```

Comment:
Is information right for this? For instance we have test infrastructure that will bail if the browser logs errors. It would be nice if this check could cause tests to fail-fast.

Reply:
Switched to error